### ISSUE TITLE
Caching file dependancies in WorkingDir

### ISSUE THREAD

**Issue by Ben8t** [kestra-io/kestra#1275]
### Feature description

Add a cache property to the `WorkingDir` task

```
type: WorkingDirectory
cache: 
- id: node
  path: nodes_modules/** 
```

**Comment by Ben8t** [comment]
from @loicmathieu :

> if you create a Scrapy task that extends AbstractPython, there is no way to set a default docker image easily.
> WHat I have in mind is that the dockerImage will be a property of the task "directly" and no more an attribute of the dockerConfig property so task that want to set a default will be able to and it will be more visible to user that wants to override it.
> ```
> tasks:
> - id: scrapy
>   type: io.kestra...Scrapy
>   runner: DOCKER
> ```
> 
> Will be enough to run Scrapy inside Docker as the task will provide the default image.
> ```
> tasks:
> - id: scrapy
>   type: io.kestra...Scrapy
>   runner: DOCKER
>   dockerImage: my_custom_scrapy_image
> ```
> WIll be enough to override the image

**Comment by loicmathieu** [comment]
Implementation idea:

- At the begining, check out internal storage and if a catch exist for this flow.task download it
- Use `pip --target` to install dependencies in a specific directory
- Setup `PYTHONPATH` to contain the dependencies directory
- At the end, ZIP the directory and put it in our internal storage (TTL ?), maybe use State for this at it must be a unique cach for a flow.task

**Comment by Ben8t** [comment]
Will probably be closed by #1245
